django/core/signing.py

```diff
@@ -158,15 +159,27 @@ def signature(self, value):
         return base64_hmac(self.salt + 'signer', value, self.key)
 
     def sign(self, value):
-        return '%s%s%s' % (value, self.sep, self.signature(value))
+        return '%s%ssha1%s%s' % (value, self.sep, self.sep, self.signature(value))
```

Comment:
Any thoughts about joining by `self.sep` instead? I find it more readable

```suggestion
        return self.sep.join([value, 'sha1', self.signature(value)])
```

Reply:
I think we should revert this change (we reverted similar change in `django.core.cache.backends.basedefault_key_func()` in the past) because it will create a regression for non string values. The following example works in the current master:
```
>>> Signer('some_key').sign(1)
'1:gJ9gvYHWvcR2rrXTSANB5b-IhU8'
```
but with this change it raises `TypeError`:
```
File "django/django/core/signing.py", line 162, in sign
    return self.sep.join([value, 'sha1', self.signature(value)])
TypeError: sequence item 0: expected str instance, int found
```

Reply:
Sure.